backend/experiment/actions/base_action.py

```diff
@@ -13,7 +13,8 @@ def action(self):
         action_dict = self.__dict__
         action_dict['view'] = self.ID
 
-        if self.style is not None:
+        # we may have already converted the style object to a dictionary, e.g., after copying an Action object
```

Comment:
Just interested: Why could this be converted to a dict already? Shouldn't serialization ideally happen at the end?

Reply:
Not if we are using copies of the same object, which speech2song does. It makes on `Trial` object, and from that creates an array of `[Trial]*n_representations`. As the serialization goes through the list of actions, it will have already converted the Playback's style object to a `dict`, so in the next step it will attempt to apply the `to_dict` method again to the `style` attribute. Took me quite a long time to understand what was going on and why.